core/src/main/java/org/infinispan/commands/read/EntrySetCommand.java

```diff
@@ -57,6 +57,10 @@ public LoadType loadType() {
 
    @Override
    public Set<CacheEntry<K, V>> perform(InvocationContext ctx) throws Throwable {
+      Object lockOwner = ctx.getLockOwner();
+      if (ctx.getLockOwner() != null) {
+         return new BackingEntrySet<>(cache.getAdvancedCache().lockAs(lockOwner));
```

Comment:
There is nothing stopping a locked stream in another and actually it would work perfectly well, unless someone changed the lock object by invoking `lockAs` themselves, which was why I was trying to tell people they can't guarantee how it would work.

This block though has nothing to do with locked stream, all it is doing is paying attention to the lock owner for the underlying stream. This allows the user to call `setValue` since that needs the appropriate lock owner inside the locked stream.